cmd/proxy/Dockerfile

```diff
@@ -8,7 +8,7 @@ WORKDIR $GOPATH/src/github.com/gomods/athens
 
 COPY . .
 
-RUN GO111MODULE=on CGO_ENABLED=0 go build -mod=vendor -o /bin/athens-proxy ./cmd/proxy
+RUN ./scripts/build.sh
```

Comment:
@robjloranger What do you think about using docker build-time variables?
We could pass the Version obtained here: https://github.com/gomods/athens/blob/master/scripts/push-docker-images.sh
So instead of https://github.com/gomods/athens/blob/f53ab925034a5c42cbf77d1bd2c53e6fa671f69b/scripts/push-docker-images.sh#L37
we could do 
`docker build --build-arg VERSION=${VERSION} -t ${REGISTRY}proxy:${VERSION} -f ${REPO_DIR}cmd/proxy/Dockerfile ${REPO_DIR}`
and use the Version in the Dockerfile
This way we wouldn't need to get the version twice.
Let me know what you think of it!

Reply:
I like it, I'll make the change. No sense doubling effort.

Reply:
All set, as a bonus now we can check the image history and see the version as well. Build args are visible there.

Reply:
not sure, but I think you need to define the variable in the Dockerfile
`ARG VERSION=default_value`

Reply:
also shouldn't it be `./scripts/build.sh ${VERSION}`? and then in `build.sh` `VERSION=$1`?
or `VERSION=${VERSION} ./scripts/build.sh`?

Reply:
I'm not certain, I'll have to look into it tonight.

Reply:
I'm going to have to do more research tomorrow. Playing with it tonight I can't get the script to execute so I'm doing something wrong there. If I run `ls` instead, I am in the directory I expect so it's not that.

Reply:
It's because it doesn't exist in `PATH`, if you ran it with `CMD` it would work. Either do that or sort the path.

eg. this works, but so would setting `ENV PATH`
```
Step 4/13 : CMD ./scripts/build.sh
 ---> Running in 78205bcd36e3
Removing intermediate container 78205bcd36e3
```

Reply:
if we ever release raw binaries, would they too have the version in them? no biggie if not, just wondering

Reply:
@arschles we use goreleaser to release raw binaries and it takes care of the version as well.

Reply:
I dropped the script and declaring `ARG VERSION` fixed the issue passing a value in. Build date is generated in the Dockerfile now so all docker builds will have a date and when we build it will have a version as well.